File: remove-element/remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int i=0;
        for(int j=0; j<nums.length; j++){
            if(nums[i]==val && nums[j]!=val){
                int temp = nums[i];
                nums[i]=nums[j];
                nums[j]=temp;
                i++;
            }else if(nums[i]!=val){
                i++;
            }
        }
        return i;
    }
}

File: 27-remove-element/remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        
        count = 0
        i = 0
        j = len(nums) - 1

        while i <=j:
            if nums[i] != val:
                count += 1
                i+= 1
            else:
                while(j>= i and nums[j] == val):
                    j -= 1
                if i < j and nums[i] == val and nums[j] != val:
                    nums[i], nums[j] = nums[j], nums[i]
        return count